tests/end-to-end/upgrade/pkg/tests/eventmesh-migration/helpers/event.go

```diff
@@ -0,0 +1,261 @@
+	transport := &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout: timeout,
+		}).DialContext,
+	}
+	client := http.Client{Transport: transport}
+	res, err := client.Post(target,
+		"application/json",
+		strings.NewReader(payload))
+	if err != nil {
+		return errors.Wrap(err, "HTTP POST request failed in SendEvent() ")
+	}
+
+	if err := verifyStatusCode(res, 200); err != nil {
+		return errors.Wrap(err, "HTTP POST request returned non-2xx failed in SendEvent() ")
+	}
+
+	return nil
+}
+
+func CheckEvent(target, eventType, eventTypeVersion string, retryOptions ...retry.Option) error {
+	return retry.Do(func() error {
+		transport := &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: timeout,
+			}).DialContext,
+		}
+		client := http.Client{Transport: transport}
+		res, err := client.Get(target)
+		if err != nil {
+			return errors.Wrap(err, "HTTP GET failed in CheckEvent()")
+		}
+
+		if err := verifyStatusCode(res, 200); err != nil {
+			return errors.Wrap(err, "HTTP GET request returned non-2xx in CheckEvent()")
+		}
+		body, err := ioutil.ReadAll(res.Body)
+		defer res.Body.Close()
+
+		if err != nil {
+			errors.Wrap(err, "failed ReadAll() in CheckEvent")
+		}
+		var resp http.Header
+		err = json.Unmarshal(body, &resp)
+		if err != nil {
+			errors.Wrap(err, "failed Unmarshal() in CheckEvent")
+		}
+
+		err = res.Body.Close()
+		if err != nil {
+			errors.Wrap(err, "failed Close() CheckEvent")
+		}
+
+		if resp.Get("ce-type") != eventType {
+			return fmt.Errorf("event type wrong: Got %s, Wanted: %s", resp.Get("ce-type"), eventType)
+		}
+		if resp.Get("ce-eventtypeversion") != eventTypeVersion {
+			return fmt.Errorf("event type version wrong: Got %s, Wanted: %s", resp.Get("ce-eventtypeversion"), eventTypeVersion)
+		}
+
+		return nil
+	}, retryOptions...)
+
+	return nil
+}
+
+type TriggerOption func(trigger *v1alpha1.Trigger)
+
+func WithURISubscriber(target string) TriggerOption {
+	url, err := apis.ParseURL(target)
+	if err != nil {
+		// TODO(k15r): proper error handling here
+		return func(trigger *v1alpha1.Trigger) {
+			return
+		}
+	}
+	destination := duckv1.Destination{
+		URI: url,
+	}
+	return func(trigger *v1alpha1.Trigger) {
+		trigger.Spec.Subscriber = destination
+	}
+}
+
+func WithFilter(eventTypeVersion, eventType, source string) TriggerOption {
+	filter := v1alpha1.TriggerFilter{
+		Attributes: &v1alpha1.TriggerFilterAttributes{
+			"eventtypeversion": eventTypeVersion,
+			"type":             eventType,
+			"source":           source,
+		},
+	}
+
+	return func(trigger *v1alpha1.Trigger) {
+		trigger.Spec.Filter = &filter
+	}
+}
+
+func CreateTrigger(eventingCli eventingclientv1alpha1.EventingV1alpha1Interface, name, namespace string, triggeroptions ...TriggerOption) error {
+	labelSelector := map[string]string{
+		"function": name,
+	}
+	trigger := &v1alpha1.Trigger{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Why do you define a TypeMeta in a Go object? TypeMeta is only necessary when serialization to/from JSON is involved.

Reply:
removed